From 3239ab3f7776ba9c58fd6c87ed33ad31d3eb42b7 Mon Sep 17 00:00:00 2001
From: Dominic Cleal <dcleal@redhat.com>
Date: Fri, 16 Dec 2011 20:22:54 +0000
Subject: [PATCH 1/2] (#11414) Save/execute changes on versions of Augeas <
 0.3.6

Versions of Augeas prior to 0.3.6 didn't report their version number, so a
fallback of executing changes once in need_to_run? and again in execute_changes
is performed.  Otherwise a save is done in need_to_run? and this is re-used in
execute_changes.

The /augeas/events/saved node is used to tell whether the latter optimisation
happened, but the return value of #match wasn't tested correctly (it's an empty
array).
---
 lib/puppet/provider/augeas/augeas.rb     |    2 +-
 spec/unit/provider/augeas/augeas_spec.rb |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/puppet/provider/augeas/augeas.rb b/lib/puppet/provider/augeas/augeas.rb
index 7214a8b..7098ebd 100644
--- a/lib/puppet/provider/augeas/augeas.rb
+++ b/lib/puppet/provider/augeas/augeas.rb
@@ -325,7 +325,7 @@ Puppet::Type.type(:augeas).provide(:augeas) do
     begin
       open_augeas
       saved_files = @aug.match("/augeas/events/saved")
-      if saved_files
+      unless saved_files.empty?
         saved_files.each do |key|
           root = resource[:root].sub(/^\/$/, "")
           saved_file = @aug.get(key).to_s.sub(/^\/files/, root)
diff --git a/spec/unit/provider/augeas/augeas_spec.rb b/spec/unit/provider/augeas/augeas_spec.rb
index 874f70a..059f5aa 100755
--- a/spec/unit/provider/augeas/augeas_spec.rb
+++ b/spec/unit/provider/augeas/augeas_spec.rb
@@ -467,7 +467,7 @@ describe provider_class do
       @augeas = stub("augeas")
       @provider.aug= @augeas
       @provider.stubs(:get_augeas_version).returns("0.3.5")
-      @augeas.stubs(:match).with("/augeas/events/saved")
+      @augeas.stubs(:match).with("/augeas/events/saved").returns([])
     end
 
     it "should handle set commands" do
-- 
1.7.7.4

